bigquery/api/Snippets/TableInsertRows.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright(c) 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+//
+// [START bigquery_table_insert_rows]
+
+using Google.Cloud.BigQuery.V2;
+using System;
+
+public class BigQueryTableInsertRows
+{
+    public void TableInsertRows(
+        string projectId = "your-project-id",
+        string datasetId = "your_dataset_id",
+        string tableId = "your_table_id"
+    )
+    {
+        BigQueryClient client = BigQueryClient.Create(projectId);
+        BigQueryInsertRow[] rows = new BigQueryInsertRow[]
+        {
+            // The insert ID is optional, but can avoid duplicate data
+            // when retrying inserts.
+            new BigQueryInsertRow(insertId: "row1") {
+                { "name", "Washington" },
+                { "post_abbr", "WA" }
+            },
+            new BigQueryInsertRow(insertId: "row2") {
+                { "name", "Colorado" },
+                { "post_abbr", "CO" }
+            }
+        };
+        client.InsertRows(datasetId, tableId, rows);
```

Comment:
Hmm... I'd expect to see checking for partial failure to insert rows, but I can't tell from the API reference how partial failures are reported.

@jskeet What happens when there are errors inserting some rows? Does[InsertRows](https://googleapis.github.io/google-cloud-dotnet/docs/Google.Cloud.BigQuery.V2/api/Google.Cloud.BigQuery.V2.BigQueryClient.html#Google_Cloud_BigQuery_V2_BigQueryClient_InsertRows_Google_Apis_Bigquery_v2_Data_TableReference_System_Collections_Generic_IEnumerable_Google_Cloud_BigQuery_V2_BigQueryInsertRow__Google_Cloud_BigQuery_V2_InsertOptions_) raise an exception?

In the other languages, the equivalent of InsertRows returns a list of errors, which need to be explicitly checked for. The reason for this is that the API method can have partial failures, where only some of the batch of rows could not be inserted.

Reply:
Will need to check. I'm at a conference for the first half of this week, so it probably won't be for a little while, but will make sure I get to it when I can.

Reply:
Right - that sounds like what I'd expect, and here's the code that checks:
https://github.com/googleapis/google-cloud-dotnet/blob/f14444bbed6056b439c2961c317f27eb7dacc3aa/apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClientImpl.InsertData.cs#L151

Anything else I need to pay attention to on this PR? (There's a lot of conversation that's tricky to keep up with while also conferencing...)